x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/FieldAttribute.java

```diff
@@ -105,6 +107,19 @@ public FieldAttribute(StreamInput in) throws IOException {
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
+        if (out.getTransportVersion().onOrAfter(TransportVersions.ESQL_FIELD_ATTRIBUTE_CACHED_SERIALIZATION)
+            && out instanceof PlanStreamOutput pso) {
```

Comment:
nit: I think this should fail if the stream output is not a `PlanStreamOutput`; the receiving side will expect the serialization to be a certain way depending on the transport version, only, right?

Similarly for `readFrom`.

Reply:
I'm replacing these checks with asserts, the tests don't seem to complain